### PR TITLE
[SDEV3-940] Enhanced logger w/ winston

### PR DESCRIPTION
**Pull request by kengoldfarb**
Backwards-compatible with log levels and usage. This should be a drop in replacement for existing projects.  However, I've marked this as a **major** release because the log format / output has been adjusted.

* Use spruce eslint config

* Uses [winston](https://github.com/winstonjs/winston) for BE logs

* Changes `useTrace` and `useSourcemaps` to default to false for performance improvements

* Add `asJSON` option to format log output as JSON

* Add ability to pass in custom `formatters` and `transports` to allow full customization of how logs are written

  * This will be integrated into workspace w/ an upcoming PR

* Basic tests of logging functionality (not exhaustive)

* Integrate `debug` module for logger debugging

* README updates

### CODE REVIEW

**Comment by justinhillsjohnson** [comment]
Let's hold on merging this baby until after we've deployed to QA today

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,6 @@
+root = true
+
+[*]
+trim_trailing_whitespace = false
```
Comment (justinhillsjohnson) [inline .editorconfig:4]: Are these updates necessary for the logger? Where do these defaults come from and are they set consistently and/or sanely across all repos?

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
+// 20170402175629
```
Comment (justinhillsjohnson) [inline .vscode/launch.json:1]: Is this file update related?

**Review by corbanb — APPROVED** [review]
LGTM :shipit:

**Review by justinhillsjohnson — CHANGES_REQUESTED** [review]
Seems like maybe we have some unrelated changes in this PR. Changes that would be good to have, but shouldn't be done as part of the logger PR. I left a couple of comments but I see other clean-up happening beyond that.

**Review by justinhillsjohnson — APPROVED** [review]

**Pull request merged**